src/http/api/resources/dht.js

```diff
@@ -0,0 +1,211 @@
+'use strict'
+
+const Joi = require('joi')
+
+const CID = require('cids')
+
+const debug = require('debug')
+const log = debug('jsipfs:http-api:dht')
+log.error = debug('jsipfs:http-api:dht:error')
+
+exports = module.exports
+
+exports.findPeer = {
+  validate: {
+    query: Joi.object().keys({
+      arg: Joi.string().required()
+    }).unknown()
+  },
+  handler: (request, reply) => {
```

Comment:
These handlers now need to change for Hapi 18 - please shout if you have any questions!